src/tests/ports/owners.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { ISubgraphComponent } from '@well-known-components/thegraph-component'
+import { FetchOptions } from '../../ports/merger/types'
+import { createOwnersComponent } from '../../ports/owner/component'
+import {
+  IOwnerDataComponent,
+  OwnerFragment,
+  OwnersFilters,
+  OwnersSortBy,
+} from '../../ports/owner/types'
+
+let collectionSubgraphMock: ISubgraphComponent
+let getOwnersQueryMock: jest.Mock
+let ownersComponentMock: IOwnerDataComponent
+let queryFilters: FetchOptions<OwnersFilters, OwnersSortBy>
+
+describe('when fetching owners', () => {
+  beforeEach(() => {
+    getOwnersQueryMock = jest.fn()
+
+    collectionSubgraphMock = {
+      query: getOwnersQueryMock,
+    }
+
+    ownersComponentMock = createOwnersComponent({
+      subgraph: collectionSubgraphMock,
+    })
+
+    queryFilters = {
+      contractAddress: 'contractAddress',
+      itemId: 'itemId',
+      first: 10,
+      skip: 10,
+      sortBy: OwnersSortBy.ISSUED_ID,
+    }
+  })
+
+  afterEach(() => {
+    getOwnersQueryMock.mockRestore()
+  })
+
+  describe('and the request fails', () => {
```

Comment:
and the request fails or and the request is successful are not meaningful as they don't provide context, they are results.

A better structure would be:

```
when fetching owners
-> when itemId is not provided
--> should throw an http 400 error with a message requiring itemId and contractAddress
-> when contractAddress is not provided
--> should throw an http 400 error with a message requiring itemId and contractAddress
-> when the subgraph component throws an error
--> should be propagated to fetchAndCount
-> when the subgraph component resolves 2 owner fragments
--> should be mapped to 2 owners
...
```

And so on